### PR TITLE
US-12: Creating Backend Using FastAPI

### PR DESCRIPTION
**Pull request by achaur11**

### CODE REVIEW

File: taigaProject/taigaApi/controller/auth_controller.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from fastapi import FastAPI, Request, APIRouter
+
+from taigaApi.auth.authenticate import authenticate
+from taigaApi.model.authRequest import AuthRequest
+
+from taigaApi.util.SimpleCache import cache
+
+router = APIRouter()
+
+@router.get("/auth")
```
Comment (shahih25) [inline taigaProject/taigaApi/controller/auth_controller.py]: It needs to be POST api as it is accepting a form.

File: taigaProject/taigaApi/controller/auth_controller.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+def auth(auth: AuthRequest):
+    token = authenticate(auth.username, auth.password)
+    cache.set("token", token)
+    return {"auth_token": token}
+
```
Comment (shahih25) [inline taigaProject/taigaApi/controller/auth_controller.py:15]: nit: remove extra lines.

File: taigaProject/taigaApi/controller/project_controller.py [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    project_info = get_project_by_slug(project_slug, token)
+    task_status_name = get_project_task_status_name(project_info["id"], token)
+
+    project_details = {
+        "name": project_info["name"],
+        "team_members": [member["full_name"] for member in project_info.get("members", [])],
+        "taskboard_column": [status["name"] for status in task_status_name],
+    }
+    return project_details
+
```
Comment (shahih25) [inline taigaProject/taigaApi/controller/project_controller.py:25]: nit: remove extra lines.

File: README.md [hunk 4]
```diff
@@ -43,3 +43,25 @@ To interact with the Taiga API using the provided Python script, you will need t
...
+After running the command 
+``` bash
+   pip install -r requirements.txt
+   ```
+Go to the taigaProject folder (`cd taigaProject`) and run the command- 
+``` bash
+uvicorn main:app 
+```
+
+You can hit the API using `http:127.0.0.1/[requiredApiPath]`
```
Comment (shahih25) [inline README.md]: nit: **http://127.0.0.1/8000/[requiredPath]** as "//" and port missing in the above.

Also, **http:127.0.0.1** is same as **http://localhost**.

File: taigaProject/taigaApi/controller/auth_controller.py (hunk 1 above)
Comment (achaur11) [reply taigaProject/taigaApi/controller/auth_controller.py]: Made the change

File: taigaProject/taigaApi/controller/project_controller.py (hunk 3 above)
Comment (achaur11) [reply taigaProject/taigaApi/controller/project_controller.py:25]: It is a considered a good practice to leave a newline at the end of a file.

File: README.md (hunk 4 above)
Comment (achaur11) [reply README.md]: Made the change

File: taigaProject/taigaApi/controller/auth_controller.py (hunk 2 above)
Comment (achaur11) [reply taigaProject/taigaApi/controller/auth_controller.py:15]: Same as below

File: taigaProject/taigaApi/controller/project_controller.py (hunk 3 above)
Comment (shahih25) [reply taigaProject/taigaApi/controller/project_controller.py:25]: There are multiple new lines from L#25 to L#29.

Comment (shahih25) [reply taigaProject/taigaApi/controller/project_controller.py:25]: We can remove in later changes.

**Review by shahih25 — APPROVED** [review]
LGTM.

**Pull request merged**